### DISCUSSION TITLE
[BUG] Unable to uninstall Longhorn using Helm: "Error: job failed: BackoffLimitExceeded"

### DISCUSSION THREAD

**Discussion by kamzata** [longhorn/longhorn#5333] [Q&A] [answered]
I receive this error whenever I try to uninstall Longhorn using Helm:

```
kamzata@k8s-master:~$ helm uninstall  longhorn -n longhorn-system
Error: job failed: BackoffLimitExceeded
```

## To Reproduce

1. Type `helm uninstall longhorn -n longhorn-system`
2. See error `Error: job failed: BackoffLimitExceeded`

## Expected behavior

Uninstall done

## Environment

 - Longhorn version: 1.4.0
 - Installation method (e.g. Rancher Catalog App/Helm/Kubectl): Helm 3.11.0
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version: Vanilla
   - Number of management node in the cluster: 1
   - Number of worker node in the cluster: 2
 - Node config
   - OS type and version: Ubuntu 22.04 Cloud-Init
   - CPU per node: 2 vCPU
   - Memory per node: 2 GB
   - Disk type(e.g. SSD/NVMe): NVMe
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): Proxmox
 - Number of Longhorn volumes in the cluster: 0

## Additional context
```
kamzata@k8s-master:~$ h status longhorn -n longhorn-system
NAME: longhorn
LAST DEPLOYED: Mon Jan 23 01:30:01 2023
NAMESPACE: longhorn-system
STATUS: uninstalling
REVISION: 1
TEST SUITE: None
NOTES:
Longhorn is now installed on the cluster!

Please wait a few minutes for other Longhorn components such as CSI deployments, Engine Images, and Instance Managers to be initialized.

Visit our documentation at https://longhorn.io/docs/
```

**Comment by weizhe0422** [comment, marked as answer]
Hi @kamzata, I think it's because `Deleting Confirmation Flag` is set to false. This setting prevents data loss from accidental uninstallation of Longhorn. 

You can check if it is false with `kubectl get settings.longhorn.io deleting-confirmation-flag` or `Setting -> Deleting Confirmation Flag` in the UI. If false, click it true and uninstall Longhorn again.

refer to. https://longhorn.io/docs/1.4.0/references/settings/#deleting-confirmation-flag

**Reply by kamzata** [reply]
You're absolute right. I tried to reinstall it, check (it was false) and change that value to true, and then the uninstall process worked. Thank you so much.

**Comment by MarvinJWendt** [comment]
If someone stumbles across this, like I did, here are the steps that solved it for me.
(Steps mentioned in latest docs did not work for me)

1. `helm rollback longhorn 1 -n longhorn-system`
2. Deploy the following resource:

```yaml
apiVersion: longhorn.io/v1beta1
kind: Setting
metadata:
  name: deleting-confirmation-flag
  namespace: longhorn-system
value: "true"
```

3. `helm uninstall longhorn -n longhorn-system`

---

Deploying the `Setting` is needed, if a previous uninstall failed, but deleted the setting. So the value cannot be changed via the mentioned kubectl commands anymore.
